src/TwigExtensions.php

```diff
@@ -27,31 +23,31 @@ public function getFunctions() {
   }
 
   /**
-   * Appends the filemtime to the given asset path to ensure cache invalidation.
+   * Appends the modification time to the given asset file path.
    *
    * @param Twig_Environment $env
    *   The current Twig environment.
-   * @param $asset_path
-   *   The given path to the asset.
-   * @param bool $is_versioned
-   *   If the asset should be versioned by appending the filemtime.
+   * @param string $path
+   *   The path to the asset file relative to the Twig environment.
+   * @param bool $add_version
+   *   Whether to append the file modification time. Defaults to TRUE.
    *
    * @return string
    */
-  public static function getAssetPath(Twig_Environment $env, $asset_path, $is_versioned = TRUE): string {
+  public static function getAssetPath(Twig_Environment $env, string $path, bool $add_version = TRUE): string {
     $globals = $env->getGlobals();
     $context_publication = $globals['contextPublication'];
     $template_path = realpath($context_publication['templateDir']);
```

Comment:
This is a custom, CUE specific configuration variable.  But Twig actually has its regular environment configuration for configuring the templates folder (at minimum the twig loaders do), which should be used instead.

Reply:
In the worst case we'd have to test every registered namespace in each registered loader.

Alternatively, we can introduce a dedicated Twig configuration variable for this package, so we don't have any dependency on anything else.

Reply:
I'm not very convinced by adding a dedicated extension per feature as we might need other extensions/functionalities for the project as well (e.g return values from macros). When creating the extension I was thinking about a place to store custom code, and that's why I named it cue-twig-extensions. This code is not a full fledged asset extension as [the one](https://symfony.com/doc/current/frontend/encore/versioning.html) mentioned by you.

Reply:
There isn't really a dependency on CUE in this functionality; it's rather a generic Twig helper.  I would recommend to limit the scope of the extension, because this simple asset helper could be reused in other projects that aren't related to CUE.

In addition, I would suggest to try putting the JavaScript port for twigjs into the same repo, because we always need both implementations to support the same functionality.  I've seen that approach in some other repos in the past (but don't remember which) and found it quite clever.

Reply:
Ok, then let's do this 👍 Should we resolve the `$globals['contextPublication'];` before merging? Also the file should be renamed as well, or?

Reply:
Should I take care of the the TwigJS port or did you start on it already?

Reply:
Yes, the following issues still need to resolved:

- Configuration/global variable: Adding a new global variable (`asset_path_document_root`), which we have to configure for CUE then? 

- The file/class name doesn't necessarily have to be renamed, `AssetExtension` would be the only sensible alternative that comes to mind. But since this is a single purpose package, `TwigExtension` is perfectly clear.

- The repo needs to be renamed after merging.

- TwigJS port.

Can you work on these todos?

Reply:
Yes, can work on those :)